packages/wrangler/src/deploy/deploy.ts

```diff
@@ -487,46 +497,52 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 			preserveFileNames: config.preserve_file_names ?? false,
 		});
 
-		const { modules, dependencies, resolvedEntryPointPath, bundleType } =
-			props.noBundle
-				? await noBundleWorker(props.entry, props.rules, props.outDir)
-				: await bundleWorker(
-						props.entry,
-						typeof destination === "string" ? destination : destination.path,
-						{
-							bundle: true,
-							additionalModules: [],
-							moduleCollector,
-							serveAssetsFromWorker:
-								!props.isWorkersSite && Boolean(props.assetPaths),
-							doBindings: config.durable_objects.bindings,
-							jsxFactory,
-							jsxFragment,
-							tsconfig: props.tsconfig ?? config.tsconfig,
-							minify,
-							legacyNodeCompat,
-							nodejsCompat,
-							define: { ...config.define, ...props.defines },
-							checkFetch: false,
-							assets: config.assets,
-							// enable the cache when publishing
-							bypassAssetCache: false,
-							services: config.services,
-							// We don't set workerDefinitions here,
-							// because we don't want to apply the dev-time
-							// facades on top of it
-							workerDefinitions: undefined,
-							// We want to know if the build is for development or publishing
-							// This could potentially cause issues as we no longer have identical behaviour between dev and deploy?
-							targetConsumer: "deploy",
-							local: false,
-							projectRoot: props.projectRoot,
-							defineNavigatorUserAgent: isNavigatorDefined(
-								props.compatibilityDate ?? config.compatibility_date,
-								props.compatibilityFlags ?? config.compatibility_flags
-							),
-						}
-				  );
+		const {
+			modules,
+			dependencies,
+			resolvedEntryPointPath,
+			bundleType,
+			...bundle
```

Comment:
```suggestion
			sourceMapPath,
			sourceMapMetadata,
```

This will save on destructuring bundle again further down and also makes it clear here what is actually being used from the result of bundling.

Reply:
This was a bit of a type hack since the return type of `noBundleWorker` doesn't include the two source map fields (so they can't be destructured here). The type of `bundle` is `{}` which is then assignable to `Partial<BundleResult>` later on.